Packs/Illumio/Integrations/IllumioCore/IllumioCore.py

```diff
@@ -22,6 +28,171 @@ def __init__(self, port):
         super().__init__(self.message)
 
 
+class InvalidSingleSelectException(Exception):
+    """Exception class for invalid single-select option."""
+
+    def __init__(self, value, argument, argument_list):
+        self.message = "{} is an invalid value for {}. Possible values are: {}.".format(value, argument, argument_list)
+        super().__init__(self.message)
+
+
+class InvalidMultiSelectException(Exception):
+    """Exception class for invalid multi-select option."""
+
+    def __init__(self, argument, argument_list):
+        self.message = "Invalid value for {}. Possible comma separated values are {}.".format(argument, argument_list)
+        super().__init__(self.message)
+
+
+''' HELPER FUNCTIONS '''
+
+
+def validate_arguments_for_traffic_analysis(port: Optional[int], policy_decisions: list, protocol: str):
+    """Validate arguments for illumio-traffic-analysis command.
+
+    Args:
+        port: Port number.
+        policy_decisions: Policy decision to include in the search result.
+        protocol: Communication protocol.
+
+    Returns:
+        str: Error if value of the argument is not valid.
+    """
+    if port < MIN_PORT or port > MAX_PORT:  # type: ignore
+        raise InvalidPortException(port)
+
+    for decision in policy_decisions:
+        if decision not in VALID_POLICY_DECISIONS:
+            raise InvalidMultiSelectException('policy_decisions', VALID_POLICY_DECISIONS)
+
+    if protocol not in VALID_PROTOCOLS:
+        raise InvalidSingleSelectException(protocol, "protocol", VALID_PROTOCOLS)
+
+
+def validate_arguments_for_illumio_virtual_service_create_command(name: str | None, port: str | None,
```

Comment:
This function name is too long. And inconsistent with other validation functions. `validate_virtual_service_args` or similar is probably enough to convey the intent.

Reply:
Will change it to` validate_virtual_service_args`